### PR TITLE
generic Vellum schema

### PR DESCRIPTION
**Pull request by snopoke**
@emord cc @millerdev 

buddy @orangejenny 

can review per commit :fish:

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_util.py [hunk 1]
```diff
@@ -155,6 +151,95 @@ def test_get_session_schema_for_simple_module_with_case(self):
...
+        })
+
+    def test_get_session_schema_advanced_form_multiple_cases(self):
+        self.factory.new_advanced_module('visit history', 'child')
+        m2, m2f0 = self.factory.new_advanced_module('visit history', 'visit')
+        self.factory.form_requires_case(m2f0, 'child')
+        self.factory.form_requires_case(m2f0, 'visit', parent_case_type='child')
+
+        schema = util.get_session_schema(m2f0)
+        import json; print json.dumps(schema, indent=4)
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_util.py]: E702 multiple statements on one line (semicolon)

File: corehq/apps/app_manager/tests/test_util.py [hunk 2]
```diff
@@ -155,6 +151,95 @@ def test_get_session_schema_for_simple_module_with_case(self):
...
+        self.module_0, m0f0 = self.factory.new_basic_module('parent', 'gold-fish')
+        self.module_1, m1f0 = self.factory.new_basic_module('child', 'guppy', parent_module=self.module_0)
+        self.factory.form_requires_case(m0f0)
+        self.factory.form_opens_case(m0f0, 'guppy', is_subcase=True)
+
+        self.factory.form_requires_case(m1f0, 'gold-fish')
+        self.factory.form_requires_case(m1f0, 'guppy', parent_case_type='gold-fish')
+
+        schema = util.get_session_schema(m1f0)
+        import json; print json.dumps(schema, indent=4)
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_util.py]: E702 multiple statements on one line (semicolon)

**Comment by emord** [comment]
:tight: I think this means it will be slightly easier to get advanced modules working as well, but I'm not positive we will handle having multiple case ids for one case type in vellum correctly

**Comment by snopoke** [comment]
Yup, advanced modules do work except for the use case you pointed out (but
that's pretty uncommon)

File: corehq/apps/app_manager/tests/test_util.py [hunk 3]
```diff
@@ -94,9 +94,11 @@ def _add_scheduler_to_form(self, form, module, form_abreviation):
 @patch('corehq.apps.app_manager.models.is_usercase_in_use', MagicMock(return_value=False))
 @patch('corehq.apps.app_manager.util.get_per_type_defaults', MagicMock(return_value={}))
 class SchemaTest(SimpleTestCase):
+    def setUp(self):
+        self.factory = AppFactory()
```
Comment (millerdev) [inline corehq/apps/app_manager/tests/test_util.py]: 👍

**Comment by millerdev** [comment]
👍 on ✅

**Pull request merged**